api/models/productDao.js

```diff
@@ -0,0 +1,124 @@
+const dbDataSource = require("./dataSource");
+
+const getProductsGenderStatusCategory = async (condition) => {
+  let gender = condition.gender.toUpperCase();
+  let status = condition.status.toUpperCase();
+  let category = condition.category.toUpperCase();
+  let productId = condition.id.toUpperCase();
+
+  console.log(gender, status, category, productId);
+
+  const genderEnum = Object.freeze({
+    MALE: 1,
+    FEMALE: 2,
+    MALEFEMALE: 3,
+  });
+  gender = genderEnum[gender];
+
+  const statusEnum = Object.freeze({
+    BEST: 1,
+    NEW: 2,
+  });
+  status = statusEnum[status];
+
+  const categoryEnum = {
+    OUTER: 1,
+    TOP: 2,
+    BOTTOM: 3,
+  };
+  category = categoryEnum[category];
```

Comment:
```suggestion
  const GenderType = Object.freeze({
    MALE: 1,
    FEMALE: 2,
    MALEFEMALE: 3,
  });
  gender = GenderEnum[gender];

  const StatusType = Object.freeze({
    BEST: 1,
    NEW: 2,
  });
  status = StatusType[status];

  const CategoryType = {
    OUTER: 1,
    TOP: 2,
    BOTTOM: 3,
  };
  category = CategoryType[category];
```
1. enum은 Pascal Case로 작성하는것이 관례입니다.
2. enum이라는 개념적인 용어보다는 Type을 써주세요.
3. 각 GenderType, StatusType, CategoryType은 함수 내 scope에 선언하는 것이 아닌, 함수 외부에 전역적으로 선언해주세요. 그래야지 다른 함수들에서도 해당 enum들을 사용할 수 있습니다.